quick-start/src/main/ui/app/components/flows-new/edit-flow/ui/step.component.ts

```diff
@@ -8,10 +9,27 @@ import { Component, Input, Output, EventEmitter, ViewChild } from '@angular/core
 })
 export class StepComponent {
   @Input() step: any;
-  constructor(
+  @Input() databases: any;
 
+  showBody = true;
+  constructor(
+    public dialog: MatDialog
   ) {}
+  toggleBody() {
+    this.showBody = !this.showBody;
+  }
+  editSettingsClicked() {
+    const dialogRef = this.dialog.open(NewStepDialogComponent, {
+      width: '600px',
+      data: {databases: this.databases}
+    });
 
-
+    dialogRef.afterClosed().subscribe(response => {
+      if (response) {
+        // TODO Add Step to backend
+        // this.flow.steps.push(response);
```

Comment:
Please ask around about the usage of service. The implementation should be down to the service level.